code/lspgen/src/lspgen_config.c

```diff
@@ -1283,7 +1291,10 @@ lspgen_read_node_config(lsdb_ctx_t *ctx, json_t *node_obj)
         if (arr && json_is_array(arr)) {
             num_arr = json_array_size(arr);
             for (idx = 0; idx < num_arr; idx++) {
-                lspgen_read_link_config(ctx, node, json_array_get(arr, idx));
+                lspgen_read_link_config(ctx, node, json_array_get(arr, idx), adj_sid);
+		if (adj_sid > 0) {
```

Comment:
Would be better to pass a reference to &adj_sid and do the increment inside lspgen_read_link_config, rather than having the post processing being optional.